app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt

```diff
@@ -446,10 +446,13 @@ open class DefaultToolbarMenu(
         }
     }
 
-    private fun getSetDefaultBrowserItem(): BrowserMenuImageText? =
-        if (BrowsersCache.all(context).isFirefoxDefaultBrowser) {
-            null
-        } else if (FxNimbus.features.defaultBrowserMessage.value().messageLocation == MessageSurfaceId.APP_MENU_ITEM) {
+    private fun getSetDefaultBrowserItem(): BrowserMenuImageText? {
+        val browsers = BrowsersCache.all(context)
+        val config = FxNimbus.features.defaultBrowserMessage.value()
+        return if (
+            config.messageLocation == MessageSurfaceId.APP_MENU_ITEM &&
+            !browsers.isFirefoxDefaultBrowser
```

Comment:
I'm not sure how `BrowsersCache.all()` could impact performace, but  maybe we could make it a bit faster if we  inline the `BrowsersCache.all(context)` call with the condition check, this way we will only call `BrowsersCache.all( )` when it's the right message? . What do you think?

```kotlin
 config.messageLocation == MessageSurfaceId.APP_MENU_ITEM &&
 !BrowsersCache.all(context).isFirefoxDefaultBrowser
```